src/callbacks/logging/checkpointer.jl

```diff
@@ -4,14 +4,18 @@
     Checkpointer(folder)
 
 Saves `learner.model` to `folder` after every [`AbstractTrainingPhase`](#).
+If `keep_top_k` is provided, only the best k models (by smallest training loss) and the latest model are kept.
 
 Use `FluxTraining.`[`loadmodel`](#) to load a model.
 """
 struct Checkpointer <: Callback
     folder
-    function Checkpointer(folder)
+    keep_top_k::Union{Integer, Nothing}
+    top_k::PriorityQueue{<:AbstractString, <:Real}
+    function Checkpointer(folder; keep_top_k::Union{Integer, Nothing}=nothing)
```

Comment:
```suggestion
    keep_top_k::Union{Int, Nothing}
    top_k::PriorityQueue{<:AbstractString, <:Real}
    function Checkpointer(folder; keep_top_k::Union{Int, Nothing}=nothing)
```
`Integer` is an abstract type.